sdk/core/azure-core/src/test/java/com/azure/core/util/serializer/FlatteningSerializerTests.java

```diff
@@ -630,6 +635,16 @@ public void canHandleEscapedProperties() throws IOException {
         Assertions.assertEquals(productDeserialized.productType, "chai");
     }
 
+    @Test
+    public void canHandleTagsOfFinalMap() throws IOException {
+        School school = new School().withTags(new Map1<>("tag.1", "value.1"));
```

Comment:
It passes if it is non-final `Map` e.g. `HashMap`

Reply:
It appears that flattening deserialization and serialization needs to become more targeted in what has that logic applied. Right now, code generation applies the `JsonFlatten` annotation at the class level and this affects all serialization properties. Instead, it should be applied in a more targeted fashion on the field/getter and setter and only affect that field/getter and setter.

Reply:
Yeah, this is a hard problem.

There is 2 thing I can think of..
1. Customize the separator ".", so codegen could choose a more obscure UTF-8 char as separator. Not a preferable solution as it does not handle the root cause.
2. Instead of flattening it at property level in JSON, flatten it at method level (i.e., keep the nested property and class, but make getter/setter able to access the nested property). But this is still hard, as the "flattened" class could be anonymous, and codegen need some big changes (and we are not even consider the flattened class in another flattened class).

In short, no good idea from me yet.